dagmc_stats/DagmcQuery.py

```diff
@@ -27,19 +49,71 @@ def get_tris(self):
         if self.meshset is None or self.meshset == self.dagmc_file.root_set:
             meshset_lst.append(self.dagmc_file.root_set)
         else:
-            dim = self.dagmc_file._my_moab_core.tag_get_data(self.dagmc_file.dagmc_tags['geom_dim'], self.meshset)[0][0]
+            dim = self.dagmc_file._my_moab_core.tag_get_data(
+                self.dagmc_file.dagmc_tags['geom_dim'], self.meshset)[0][0]
             # get triangles of a volume
             if dim == 3:
-                surfs = self.dagmc_file._my_moab_core.get_child_meshsets(self.meshset)
+                surfs = self.dagmc_file._my_moab_core.get_child_meshsets(
+                    self.meshset)
                 meshset_lst.extend(surfs)
             # get triangles of a surface
             elif dim == 2:
                 meshset_lst.append(self.meshset)
             else:
-                warnings.warn('Meshset is not a volume nor a surface! Rootset will be used by default.')
+                warnings.warn(
+                    'Meshset is not a volume nor a surface! Rootset will be used by default.')
                 meshset_lst.append(self.dagmc_file.root_set)
 
         for item in meshset_lst:
-            tris = self.dagmc_file._my_moab_core.get_entities_by_type(item, types.MBTRI)
+            tris = self.dagmc_file._my_moab_core.get_entities_by_type(
+                item, types.MBTRI)
             tris_lst.extend(tris)
-        return tris
+        return tris_lst
+
+    def calc_tris_per_vert(self, ignore_zero=True):
+        """
+        calculate triangle per vertex data
+
+        inputs
+        ------
+        ignore_zero : (boolean) whether or not to ignore zero tris_per_vert values
+
+        outputs
+        -------
+        none
+        """
+        t_p_v_data = []
+        tri_dimension = 2
+        meshset_lst = []
+        verts = []
+        if self.meshset is None or self.meshset == self.dagmc_file.root_set:
+            meshset_lst.append(self.dagmc_file.root_set)
+        else:
+            dim = self.dagmc_file._my_moab_core.tag_get_data(
+                self.dagmc_file.dagmc_tags['geom_dim'], self.meshset)[0][0]
+            if dim == 3:
+                surfs = self.dagmc_file._my_moab_core.get_child_meshsets(
+                    self.meshset)
+                meshset_lst.extend(surfs)
+            elif dim == 2:
+                meshset_lst.append(self.meshset)
+            else:
+                warnings.warn(
+                    'Meshset is not a volume nor a surface! Rootset will be used by default.')
+                meshset_lst.append(self.dagmc_file.root_set)
```

Comment:
It seems like we need a method (internal?) to convert the `meshset` provided by the user into a valid `meshset_lst` - then we can reuse the code here and in `get_tris`

Alternatively, we could have an internal `get_entities` that takes an entity type and a meshset that gets converted to a valid meshset_lst